### PR TITLE
docs: rework ql training homepage (sd-3764)

### PR DESCRIPTION
**Pull request by jf205**
This PR makes a few changes to the 'QL training and variant analysis` homepage.
- It adds links to all of the C/C++ and Java slides that will be available, once published on help.semmle.com.
- The `training` project only had one document: index.rst. That seemed silly, so I have moved that over to the `learn-ql` project and renamed it `ql-training.rst`.
- The `ql-training-rst` project (containing the rst source files for the slide shows) has been renamed `ql-training`.
- There has been some minor rearranging of `learn-ql/index.rst` and `learn-ql/writing-queries/writing-queries.rst` 

I have also mentioned the requirement to have Sphinx 1.7.9 in order to build the slides in `readme.rst`.

It has been noted that several redirects will be required when the project structure changes make it to help.semmle.com. There will also need to be changes to `build` in the code repo.

@felicity-semmle I think this reorganisation makes sense.
@shati-semmle would you mind reviewing the text changes please? The preview is [here](http://docteam.internal.semmle.com/james/sd-3764/learn-ql/ql-training.html).

### CODE REVIEW

**Comment by jf205** [comment]
Obviously the code repo is required before generate-sphinx works on this PR. I've raised a PR for that change.

File: docs/language/learn-ql/ql-training.rst [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+- `Introduction to local data flow <../ql-training/cpp/data-flow-cpp.html>`__–an introduction to analyzing local data flow in C/C++ using QL, including an example demonstrating how to develop a query to find a real CVE.
+- `Exercise: snprintf overflow <../ql-training/cpp/snprintf.html>`__–an example demonstrating how to develop a data flow query.
+- `Introduction to global data flow <../ql-training/cpp/global-data-flow-cpp.html>`__–an introduction to analyzing global data flow in C/C++ using QL.
+- `Analyzing control flow: QL for C/C++  <../ql-training/cpp/control-flow-cpp.html>`__–an introduction to analyzing control flow in C/C++ using QL.
+
+QL and variant analysis for Java
+--------------------------------
+
+- `Introduction to variant analysis: QL for Java <../ql-training/java/intro-ql-java.html>`__–an introduction to variant analysis and QL for Java programmers.
+- `Example: Query injection <../ql-training/java/query-injection-java.html>`__–an example of iterative query development to find unsanitized SPARQL injection in a Java project.
```
Comment (shati-semmle) [inline docs/language/learn-ql/ql-training.rst]:
```suggestion
- `Example: Query injection <../ql-training/java/query-injection-java.html>`__–an example of iterative query development to find unsanitized SPARQL injections in a Java project.
```

**Comment by jf205** [comment]
Thanks for the suggestions @shati-semmle! Both accepted.

Re. the other points: Yes, I'll fix the horizontal squashing in a follow-up PR. I at least get a horizontal scroll bar on the notes that overflow. Do you see that too? 'Easily bitten' is not a typo, but also not the best phrase to use here. I'll update that!

**Comment by shati-semmle** [comment]
Ah yes, I hadn't spotted the scroll bar! 

(And sorry, the typo I was referring to was "_it_ easy to get bitten" instead of "_it's_ ...")

**Comment by jf205** [comment]
Ah, I'll fix the typo on this PR then!

**Pull request merged**